tests/Unit/ParallelAlgorithms/Amr/Criteria/Test_DriveToTarget.cpp

```diff
@@ -0,0 +1,169 @@
+#include "Domain/Amr/Flag.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Domain/Structure/SegmentId.hpp"
+#include "Framework/TestCreation.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Helpers/DataStructures/DataBox/TestHelpers.hpp"
+#include "Options/Protocols/FactoryCreation.hpp"
+#include "Parallel/GlobalCache.hpp"
+#include "Parallel/RegisterDerivedClassesWithCharm.hpp"
+#include "ParallelAlgorithms/Amr/Criteria/Criterion.hpp"
+#include "ParallelAlgorithms/Amr/Criteria/DriveToTarget.hpp"
+#include "ParallelAlgorithms/Amr/Criteria/Tags/Criteria.hpp"
+#include "Utilities/MakeArray.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace {
+template <size_t VolumeDim>
+struct Metavariables {
+  static constexpr size_t volume_dim = VolumeDim;
+  using component_list = tmpl::list<>;
+  using const_global_cache_tags = tmpl::list<>;
+  struct factory_creation
+      : tt::ConformsTo<Options::protocols::FactoryCreation> {
+    using factory_classes = tmpl::map<tmpl::pair<
+        amr::Criterion, tmpl::list<amr::Criteria::DriveToTarget<VolumeDim>>>>;
+  };
+};
+struct TestComponent {};
+
+template <size_t VolumeDim>
+void test_criterion(
+    const amr::Criterion& criterion,
+    const std::array<size_t, VolumeDim> mesh_extents,
+    const std::array<size_t, VolumeDim>& element_refinement_levels,
+    const std::array<amr::domain::Flag, VolumeDim>& expected_flags) {
+  Parallel::GlobalCache<Metavariables<VolumeDim>> empty_cache{};
+  const auto databox = db::create<tmpl::list<::domain::Tags::Mesh<VolumeDim>>>(
+      Mesh<VolumeDim>{mesh_extents, Spectral::Basis::Legendre,
+                      Spectral::Quadrature::GaussLobatto});
+  ObservationBox<tmpl::list<>,
+                 db::DataBox<tmpl::list<::domain::Tags::Mesh<VolumeDim>>>>
+      box{databox};
+
+  std::array<SegmentId, VolumeDim> segment_ids;
+  alg::transform(element_refinement_levels, segment_ids.begin(),
+                 [](const size_t extent) {
+                   return SegmentId{extent, 0_st};
+                 });
+  ElementId<VolumeDim> element_id{0, segment_ids};
+  auto flags = criterion.evaluate(box, empty_cache, element_id);
+  CHECK(flags == expected_flags);
+}
+
+template <size_t VolumeDim>
+void test(
+    const std::array<size_t, VolumeDim>& target_extents,
+    const std::array<size_t, VolumeDim>& target_levels,
+    const std::array<amr::domain::Flag, VolumeDim>& flags_at_target,
+    const std::vector<
+        std::tuple<std::array<size_t, VolumeDim>, std::array<size_t, VolumeDim>,
+                   std::array<amr::domain::Flag, VolumeDim>>>& test_cases,
+    const std::string& option_string) {
+  Parallel::register_factory_classes_with_charm<Metavariables<VolumeDim>>();
+
+  const amr::Criteria::DriveToTarget criterion{target_extents, target_levels,
+                                               flags_at_target};
+  const auto criterion_from_option_string =
+      TestHelpers::test_creation<std::unique_ptr<amr::Criterion>,
```

Comment:
I think we have a test helper that creates the metavars for you so you don't have to define them in this test. `test_factory_creation` or something like that.

Reply:
The metavariables are used in the base class so I left this as is since I need to define them